app/services/claimToAdjustPoa/RecalculatePoaHelper.scala

```diff
@@ -44,16 +47,32 @@ trait RecalculatePoaHelper extends ClientConfirmedController with FeatureSwitchi
   }
 
   private def handlePoaAndOtherData(poa: PaymentOnAccountViewModel,
-                                      otherData: PoAAmendmentData, isAgent: Boolean, nino: Nino, ctaCalculationService: ClaimToAdjustPoaCalculationService)
+                                      otherData: PoAAmendmentData, isAgent: Boolean, nino: Nino, ctaCalculationService: ClaimToAdjustPoaCalculationService, auditingService: AuditingService)
                                      (implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext,
                                       itvcErrorHandler: ItvcErrorHandler, itvcErrorHandlerAgent: AgentItvcErrorHandler): Future[Result] = {
     otherData match {
       case PoAAmendmentData(Some(poaAdjustmentReason), Some(amount), _) =>
         ctaCalculationService.recalculate(nino, poa.taxYear, amount, poaAdjustmentReason) map {
           case Left(ex) =>
             Logger("application").error(s"POA recalculation request failed: ${ex.getMessage}")
+            auditingService.extendedAudit(AdjustPaymentsOnAccountAuditModel(
+              isSuccessful = false,
+              previousPaymentOnAccountAmount = poa.totalAmountOne,
+              requestedPaymentOnAccountAmount = amount,
+              adjustmentReasonCode = poaAdjustmentReason.code,
+              adjustmentReasonDescription = Messages(poaAdjustmentReason.messagesKey),
```

Comment:
as I understand you are planning to switch from English to Welsh, depending on the context?
we only use one single language in audit events;

Reply:
The Adjustment reason text only appears in the messages file.
I didn't want to hard code the text for the adjustment reason so decided to use the case objects in this file: https://github.com/hmrc/income-tax-view-change-frontend/blob/main/app/models/claimToAdjustPoa/SelectYourReason.scala 
All the descriptions use a key from the messages file.

Reply:
* This would not work, I've just tested this locally;
* There will be switch to use Welsh if user selected Welsh, we expect all audit messages to be in English;

<img width="2088" alt="Screenshot 2024-10-08 at 13 41 16" src="https://github.com/user-attachments/assets/67f86f96-b75c-4681-9d76-230eb424ff87">
<img width="989" alt="Screenshot 2024-10-08 at 13 40 19" src="https://github.com/user-attachments/assets/bd4506d6-dd80-4144-ae08-bf56d51560d8">

Reply:
Ok, let me fix that.
Looks like I need to rebase as well!!!